pkg/controller/jobframework/reconciler.go

```diff
@@ -473,16 +475,35 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 	return match, nil
 }
 
+func FindMatchingWorkloads(ctx context.Context, c client.Client, job GenericJob, object client.Object) (match *kueue.Workload, toDelete []*kueue.Workload, workloads *kueue.WorkloadList, err error) {
+	workloads = &kueue.WorkloadList{}
+	if err := c.List(ctx, workloads, client.InNamespace(object.GetNamespace()),
+		client.MatchingFields{getOwnerKey(job.GVK()): object.GetName()}); err != nil {
+		return nil, nil, nil, err
+	}
+
+	for i := range workloads.Items {
+		w := &workloads.Items[i]
+		if match == nil && equivalentToWorkload(job, object, w) {
+			match = w
+		} else {
+			toDelete = append(toDelete, w)
+		}
+	}
+
+	return match, toDelete, workloads, nil
+}
+
 // equivalentToWorkload checks if the job corresponds to the workload
-func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Object, wl *kueue.Workload) bool {
+func equivalentToWorkload(job GenericJob, object client.Object, wl *kueue.Workload) bool {
 	owner := metav1.GetControllerOf(wl)
 	// Indexes don't work in unit tests, so we explicitly check for the
 	// owner here.
 	if owner.Name != object.GetName() {
```

Comment:
this could be `owner.Name != job.Object().GetName()`, so that we don't need the extra `object` parameter. Having the extra parameter makes it seem that `job` and `object` are not related.